sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -0,0 +1,437 @@
+        super(pipeline, url, serviceVersion, accountName, fileSystemName, fileName, blockBlobAsyncClient);
+    }
+
+    DataLakeFileAsyncClient(PathAsyncClient pathAsyncClient) {
+        super(pathAsyncClient.getHttpPipeline(), pathAsyncClient.getPathUrl(), pathAsyncClient.getServiceVersion(),
+            pathAsyncClient.getAccountName(), pathAsyncClient.getFileSystemName(), pathAsyncClient.getObjectPath(),
+            pathAsyncClient.getBlockBlobAsyncClient());
+    }
+
+    /**
+     * Gets the URL of the file represented by this client on the Data Lake service.
+     *
+     * @return the URL.
+     */
+    public String getFileUrl() {
+        return getPathUrl();
+    }
+
+    /**
+     * Gets the path of this file, not including the name of the resource itself.
+     *
+     * @return The path of the file.
+     */
+    public String getFilePath() {
+        return getObjectPath();
+    }
+
+    /**
+     * Gets the name of this file, not including its full path.
+     *
+     * @return The name of the file.
+     */
+    public String getFileName() {
+        return getObjectName();
+    }
+
+    /**
+     * Creates a file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.create}
+     *
+     * <p>For more information see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create">Azure
+     * Docs</a></p>
+     *
+     * @return A reactive response containing information about the created file.
+     */
+    public Mono<PathInfo> create() {
+        try {
+            return createWithResponse(null, null, null, null, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Creates a file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.createWithResponse#PathHttpHeaders-Map-DataLakeRequestConditions-String-String}
+     *
+     * <p>For more information see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create">Azure
+     * Docs</a></p>
+     *
+     * @param headers {@link PathHttpHeaders}
+     * @param metadata Metadata to associate with the resource.
+     * @param accessConditions {@link DataLakeRequestConditions}
+     * @param permissions POSIX access permissions for the file owner, the file owning group, and others.
+     * @param umask Restricts permissions of the file to be created.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains a {@link
+     * PathItem}.
+     */
+    public Mono<Response<PathInfo>> createWithResponse(PathHttpHeaders headers, Map<String, String> metadata,
```

Comment:
>createWithResponse [](start = 36, length = 18)

Should we have overwrite behavior similar to blobs where we don't overwrite by default. I don't know what ADLS customers would expect and if this would be really unintuitive for them, but we should consider.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/6104